dask/dataframe/io/sql.py

```diff
@@ -211,3 +213,173 @@ def _read_sql_chunk(q, uri, meta, engine_kwargs=None, **kwargs):
         return meta
     else:
         return df.astype(meta.dtypes.to_dict(), copy=False)
+
+
+def to_sql(
+    df,
+    name: str,
+    uri: str,
+    schema=None,
+    if_exists: str = "fail",
+    index: bool = True,
+    index_label=None,
+    chunksize=None,
+    dtype=None,
+    compute=True,
+    parallel=False,
+    **kwargs
```

Comment:
Just accept a method keyword here regardless of the version. Then if pandas < 0.24.0 you can raise an exception if the default is not provided.

Then we don't have to validate that we aren't silently discarding arguments.